### PR TITLE
Storybook Addon Fixes

### PR DESCRIPTION
**Pull request by adierkens**
CleanShot 2024-08-01 at 00 32 31

Requires: https://github.com/player-ui/rules_player/pull/63
Will let us remove/deprecate the storybook-webpack plugin [here](https://github.com/player-ui/tools/tree/main/xlr/asset-docgen-webpack-plugin)

## Release Notes
- Re-adds the ability to render Asset properties as a tab in storybook
- Re-adds the flow-refresh addon
- Fixes the dependencies & package layout for the storybook addon
- Fix dark-mode support

<details>
  <summary>📦 Published PR as canary version: <code>0.7.5--canary.449.15808</code></summary>
  <br />

  Try this version out locally by upgrading relevant packages to 0.7.5--canary.449.15808
</details>

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -5,13 +5,18 @@ module(
...
-  strip_prefix = "rules_player-1.0.1",
-  urls = ["https://github.com/player-ui/rules_player/archive/refs/tags/v1.0.1.tar.gz"],
-  integrity = "sha256-zIbxx1Bh/X4E8OwbPJz1nY9mOd2uAgmTZyf1Hs2mq+U="
+# archive_override(
+#   module_name = "rules_player",
+#   strip_prefix = "rules_player-1.0.1",
+#   urls = ["https://github.com/player-ui/rules_player/archive/refs/tags/v1.0.1.tar.gz"],
+#   integrity = "sha256-zIbxx1Bh/X4E8OwbPJz1nY9mOd2uAgmTZyf1Hs2mq+U="
+# )
+git_override(
```
Comment (KetanReddy) [inline MODULE.bazel:14]: We'll probably want the rules to be published before this gets merged

File: tools/storybook/src/addons/docs/index.tsx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+  }
+
+  if (node.type === "object" && node.name) {
+    return node.name;
+  }
+
+  return node.type;
+}
+
+function XLRObjectDocs(props: {
```
Comment (KetanReddy) [inline tools/storybook/src/addons/docs/index.tsx:49]: Moving this here would be great. We'll probably need to figure out how we want to go about deprecating the webpack xlr plugin

Comment (adierkens) [reply tools/storybook/src/addons/docs/index.tsx:49]: Is there anything that would stop us from just not publishing new versions? It wouldn't work with the updated storybook plugin updates for v7 anyway

**Review by KetanReddy — APPROVED** [review]

File: tools/storybook/src/addons/docs/index.tsx (hunk 2 above)
Comment (adierkens) [reply tools/storybook/src/addons/docs/index.tsx:49]: The usage instructions were already updated on the storybook page: https://github.com/player-ui/player/pull/449/files#diff-bb7fbe3850070f90366f286656bbc59ebfb8a8734d46fd8796cd1738f99d40e1R71

**Pull request merged**